src/osd/OSDMap.cc

```diff
@@ -2149,20 +2149,20 @@ int OSDMap::apply_incremental(const Incremental &inc)
   if (inc.new_full_ratio >= 0) {
     full_ratio = inc.new_full_ratio;
   }
-  if (inc.new_require_min_compat_client > 0) {
+  if (inc.new_require_min_compat_client != ceph_release_t::unknown) {
     require_min_compat_client = inc.new_require_min_compat_client;
   }
-  if (inc.new_require_osd_release >= 0) {
+  if (inc.new_require_osd_release != ceph_release_t::unknown) {
```

Comment:
this changes the encoding of osdmap.

http://pulpito.ceph.com/kchai-2019-05-03_02:49:24-rados-wip-kefu-testing-2019-05-01-1108-distro-basic-smithi/3920053